spot_driver/launch/spot_driver.launch.py

```diff
@@ -2,51 +2,134 @@
+            PathJoinSubstitution([pkg_share, "urdf", "spot.urdf.xacro"]),
+            " ",
+            "arm:=",
+            has_arm,
+            " ",
+            "tf_prefix:=",
+            tf_prefix,
+            " ",
+        ]
+    )
 
-    params = {"robot_description": robot_desc}
+    params = {"robot_description": robot_description}
     robot_state_publisher = launch_ros.actions.Node(
         package="robot_state_publisher",
         executable="robot_state_publisher",
         output="screen",
         parameters=[params],
+        namespace=spot_name,
     )
     ld.add_action(robot_state_publisher)
 
+    if not rviz_config_filename:
+        create_rviz_config(spot_name)
+        rviz_config_file = PathJoinSubstitution([FindPackageShare("spot_driver"), "rviz", "spot.rviz"])
+    else:
+        rviz_config_file = PathJoinSubstitution([FindPackageShare("spot_driver"), "rviz", rviz_config_filename])
+
+    rviz = launch_ros.actions.Node(
+        package="rviz2",
+        executable="rviz2",
+        name="rviz2",
+        arguments=["-d", rviz_config_file.perform(context)],
+        output="screen",
+        condition=IfCondition(launch_rviz),
+    )
+
+    ld.add_action(rviz)
+
 
 def generate_launch_description() -> launch.LaunchDescription:
-    config_file = LaunchConfiguration("config_file", default="")
-    config_file_arg = DeclareLaunchArgument(
-        "config_file", description="Path to configuration file for the driver.", default_value=""
+    launch_args = []
+
+    launch_args.append(
+        DeclareLaunchArgument(
+            "config_file",
+            default_value="",
+            description="Path to configuration file for the driver.",
+        )
     )
 
-    has_arm = LaunchConfiguration("has_arm")
-    has_arm_arg = DeclareLaunchArgument("has_arm", description="Whether spot has arm", default_value="False")
+    launch_args.append(DeclareLaunchArgument("has_arm", default_value="False", description="Whether spot has arm"))
 
-    ld = launch.LaunchDescription([config_file_arg, has_arm_arg])
+    launch_args.append(
+        DeclareLaunchArgument(
+            "tf_prefix",
+            default_value='""',
+            description="apply namespace prefix to robot links and joints",
+        )
+    )
 
-    spot_driver_node = launch_ros.actions.Node(
-        package="spot_driver", executable="spot_ros2", name="spot_ros2", output="screen", parameters=[config_file]
+    launch_args.append(DeclareLaunchArgument("launch_rviz", default_value="False", description="Launch RViz?"))
+    launch_args.append(
+        DeclareLaunchArgument(
+            "rviz_config_filename",
+            default_value="",
+            description="RViz config file name",
+        )
     )
-    ld.add_action(spot_driver_node)
 
-    ld.add_action(OpaqueFunction(function=launch_robot_state_publisher, args=[has_arm, ld]))
+    launch_args.append(DeclareLaunchArgument("spot_name", default_value="", description="Name of Spot"))
```

Comment:
When I set `tf_prefix:=<spot_name>/` and `spot_name:=<spot_name>` and launch with rviz, this happens:
![image](https://github.com/bdaiinstitute/spot_ros2/assets/137220849/cda87cbd-f990-441f-991b-d59192053c7c)
It appears that several of the links are improperly named:
![image](https://github.com/bdaiinstitute/spot_ros2/assets/137220849/1e7edbfe-8543-4ebb-bc58-2bc3b31a01b7)
I would hazard a guess that somewhere, it's not updating the link names correctly to reflect the changes to the `tf_prefix`.

Reply:
Yeah, I realize I made an incorrect assumption here and that the joint states, and consequently the joint names, are published via `spot_ros2` and so the tf_prefix will break the mapping and should be removed